etcdctl/ctlv2/command/util.go

```diff
@@ -156,7 +156,12 @@ func getTransport(c *cli.Context) (*http.Transport, error) {
 		CertFile: certfile,
 		KeyFile:  keyfile,
 	}
-	return transport.NewTransport(tls, defaultDialTimeout)
+
+	to := c.GlobalDuration("total-timeout")
```

Comment:
no. 
Your code has two issues:

1. we should never modify the default variable in any case.
2. we should keep the shorter timeout not longer. Say I want the total timeout to be 5 seconds, but the defaultDialtimeout is 30 seconds. I should give the transport a timeout of 5 seconds, not 30 seconds. Or dial might block for 30 seconds, which is longer than 5 seconds.

```go
    dailTimeout := defaultDialTimeout 
    totalTimeout := c.GlobalDuration("total-timeout")
    if totalTimoeout != 0 && totalTimeout < defaultDialTimeout {
        defaultDialTimeout = totalTimeout
    }
    return transport.NewTransport(tls, defaultDialTimeout)
```